agent/agent.go

```diff
@@ -546,9 +552,9 @@ func customSeekers(cfg *ProductConfig, tmpDir string) ([]*seeker.Seeker, error)
 	var err error
 	switch cfg.Name {
 	case product.Consul:
-		c = client.NewConsulAPI()
+		c, err = client.NewConsulAPI()
```

Comment:
Ensure err is handled

Reply:
I think it should be handled on [line 563](https://github.com/hashicorp/hcdiag/pull/103/files/896abcf9658fda690a0fd9dfeb6e0887f4aaca29#diff-62b6ad581fe3a3059ae8c85ef0f31dde4092bfdecfa7d6857c470bcacaa8cc8bR563).

Reply:
I believe that if the product config isn't set then branch would get skipped. This is a good example of a gotcha to watch out for in go errors, where assigning to the same `err` var multiple times can cause errors to get overwritten and lost without being explicitly handled.

Reply:
I'm afraid I'm missing something. 😟  Since go only evaluates one case (and doesn't need the explicit `break` from C), I thought only one case would be hit, so err would only be assigned one value. I could see issues potentially starting around line 567 if none of the above cases had been hit, but I didn't touch that code in this PR. Also, since I didn't touch `client.NewTFEAPI`, I did not update that case either.
<img width="483" alt="image" src="https://user-images.githubusercontent.com/16619375/164742455-6a683ff9-6bf8-406a-8620-e305d41f3d10.png">

Reply:
Oh, that's my bad! code indentation threw me off a bit there, i thought that we were relying on error handling in the vault clause. here's what the diff view looked like for 63:

![Screen Shot 2022-04-22 at 11 46 47 AM](https://user-images.githubusercontent.com/938395/164776036-7b7a5c18-c44a-456b-81d2-07fcb9df119e.png)

Reply:
Oh, okay, cool! Glad I wasn't missing something! 😄